datafusion/spark/src/function/string/length.rs

```diff
@@ -279,4 +286,38 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_spark_length_nullability() -> Result<()> {
+        let func = SparkLengthFunc::new();
+
+        let nullable_field: FieldRef =
+            Arc::new(Field::new("col", DataType::Utf8, true));
+
+        let out_nullable = func.return_field_from_args(ReturnFieldArgs {
+            arg_fields: &[nullable_field],
+            scalar_arguments: &[], 
```

Comment:
there should be the same number of `scalar_arguments` and `arg_fields`, if something is not scalar than None is used.
```suggestion
            scalar_arguments: &[None], 
```